tests/src/serverTestArrayPersistence.cpp

```diff
@@ -0,0 +1,150 @@
+    boost::filesystem::create_directory("hist");
+
+    std::vector<int> vint(alen);
+    std::vector<double> vdouble(alen);
+    for(int i = 0; i < alen; i++) {
+      vint[i] = 100 + i;
+      vdouble[i] = 200.1 + i;
+    }
+    vectorToFile(vint, "hist/INT-TO_DEVICE_ARRAY.intarray");
+    vectorToFile(vdouble, "hist/DOUBLE-TO_DEVICE_ARRAY.doublearray");
+  }
+
+  ~GlobalFixture() { GlobalFixture::referenceTestApplication.releaseManualLoopControl(); }
+
+  static ReferenceTestApplication referenceTestApplication;
+  static std::string rpcNo;
+  static std::string bpn;
+  ThreadedDoocsServer server{boost::unit_test::framework::master_test_suite().p_name.value + ".conf",
+      boost::unit_test::framework::master_test_suite().argc, boost::unit_test::framework::master_test_suite().argv};
+};
+
+ReferenceTestApplication GlobalFixture::referenceTestApplication{BOOST_STRINGIZE(BOOST_TEST_MODULE), alen};
+std::string GlobalFixture::rpcNo;
+std::string GlobalFixture::bpn;
+
+CTK_BOOST_GLOBAL_FIXTURE(GlobalFixture)
+
+// the array must have testStartValue+i at index i.
+template<class T>
+static bool testArrayContent(std::vector<T>& array, std::string const& propertyName, T testStartValue, T delta) {
+  bool isOK = true;
+  T currentTestValue = testStartValue;
+  size_t index = 0;
+  for(auto val : array) {
+    if(std::fabs(val - currentTestValue) > 0.001) {
+      if(isOK) {
+        std::cout << "Array " << propertyName << " does not contain expected values. First mismatching data at index "
+                  << index << ": " << val << " != " << currentTestValue << std::endl;
+      }
+      isOK = false;
+    }
+    currentTestValue += delta;
+    ++index;
+  }
+  return isOK;
+}
+template<class T>
+static bool testArrayContentFromProperty(std::string const& propertyName, T testStartValue, T delta) {
+  auto array = DoocsServerTestHelper::doocsGetArray<T>(propertyName);
+  return testArrayContent(array, propertyName, testStartValue, delta);
+}
+
+/// Using prepared persistence file, start server, check restored values, change values, trigger persisting, compare file with expectation
+BOOST_AUTO_TEST_CASE(testArrayWrite) {
+  GlobalFixture::referenceTestApplication.runMainLoopOnce();
+
+  std::string const& propertyAddress = "//INT/FROM_DEVICE_ARRAY";
+  checkDataType(propertyAddress, DATA_A_INT);
+
+  CHECK_WITH_TIMEOUT(testArrayContentFromProperty<float>("//INT/FROM_DEVICE_ARRAY", 100, 1) == true);
+  CHECK_WITH_TIMEOUT(testArrayContentFromProperty<float>("//DOUBLE/FROM_DEVICE_ARRAY", 200.1, 1) == true);
+
+  std::vector<int> vint(alen);
+  std::vector<double> vdouble(alen);
+  for(int i = 0; i < alen; i++) {
+    vint[i] = 150 + i;
+    vdouble[i] = 250.3 + i;
+  }
+  DoocsServerTestHelper::doocsSet("//INT/TO_DEVICE_ARRAY", vint);
+  DoocsServerTestHelper::doocsSet("//DOUBLE/TO_DEVICE_ARRAY", vdouble);
+
+  // run the application loop.
+  GlobalFixture::referenceTestApplication.runMainLoopOnce();
+
+  // check changed values
+  usleep(100000);
```

Comment:
In principle you don't need the sleep here. That's why CHECK_WITH_TIMEOUT  is trying so long.